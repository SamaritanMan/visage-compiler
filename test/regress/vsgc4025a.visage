/*
 * Regression test 
 * VSGC-4025 : Duplicate and incorrect invalidations on bound if sequence
 *
 * @test
 * @run
 */

var n = 1;
def bs = bind if(n>2) [0..n] else [0..n*2] on replace [a..b] = newVal { println("[{a}..{b}] = {sizeof newVal}") };
n = 10;
n = 1;
