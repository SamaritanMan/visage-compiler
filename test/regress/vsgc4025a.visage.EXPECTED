[0..-1] = 3
[0..2] = 11
[0..10] = 3
